web_ui/oauth2_client.go

```diff
@@ -275,6 +307,13 @@ func generateUserGroupInfo(userInfo map[string]interface{}, idToken map[string]i
 		err = errors.Errorf("invalid group source: %s", groupSource)
 		return "", nil, err
 	}
+
+	// now that we have verified that the user belongs to a group we should create the user if it doesn't exist
+	_, err = database.GetOrCreateUser(database.ServerDatabase, user, sub, issuerClaimValue)
+	if err != nil {
+		return "", nil, err
+	}
+
```

Comment:
We should create the user no matter what. That gives us more potential to have the user in our db when we go to add them to a group, making that action easier.

If they are already users then they should be enhanced to include the newly found sub + iss fields.